src/Compilers/CSharp/Portable/Symbols/Wrapped/WrappedTypeParameterSymbol.cs

```diff
@@ -109,6 +117,14 @@ public override bool HasValueTypeConstraint
             }
         }
 
+        public override bool IsValueTypeFromConstraintTypes
+        {
+            get
+            {
+                return _underlyingTypeParameter.IsValueTypeFromConstraintTypes || CalculateIsValueTypeFromConstraintTypes(ConstraintTypesNoUseSiteDiagnostics);
```

Comment:
>|| CalculateIsValueTypeFromConstraintTypes(ConstraintTypesNoUseSiteDiagnostics) [](start = 79, length = 79)

Is this necessary?

Reply:
> Is this necessary?

Yes, due to possible type substitution involved new types can appear in the constraint types

---
In reply to: [521607714](https://github.com/dotnet/roslyn/pull/49261#discussion_r521607714) [](ancestors = 521607714)